eda/src/autogluon/eda/analysis/interaction.py

```diff
@@ -0,0 +1,111 @@
+from typing import List, Optional
+
+import phik  # noqa - required for significance_matrix instrumentation on pandas dataframes
+
+from .base import AbstractAnalysis
+from .. import AnalysisState
+
+__all__ = ["Correlation", "CorrelationSignificance"]
+
+
+class Correlation(AbstractAnalysis):
+    """
+    Correlation analysis.
+
+    Parameters
+    ----------
+    method: str  {'pearson', 'kendall', 'spearman', 'phik'}, default='spearman'
+        Method of correlation:
+            * pearson : standard correlation coefficient
+            * kendall : Kendall Tau correlation coefficient
+            * spearman : Spearman rank correlation
+            * phik : phi_k correlation
+                Correlation matrix of bivariate gaussian derived from chi2-value Chi2-value
+                gets converted into correlation coefficient of bivariate gauss with correlation
+                value rho, assuming giving binning and number of records. Correlation coefficient
```

Comment:
I'm not sure what `assuming giving binning` means. Perhaps you meant `assuming given binning`?